sky/clouds/gcp.py

```diff
@@ -287,9 +294,18 @@ def get_image_size(self, image_id: str, region: Optional[str]) -> float:
                 raise ValueError(f'Image {image_id!r} not found in GCP.')
             project = image_attrs[1]
             image_name = image_attrs[-1]
-            image_infos = compute.images().get(project=project,
-                                               image=image_name).execute()
-            return float(image_infos['diskSizeGb'])
+            if self._is_machine_image(image_id):
+                image_infos = compute.machineImages().get(
```

Comment:
Probably should add an inline comment here about: We support both GCP's Machine Images and Custom Images, both of which are specified with the `image_id` field.

Should we also add to https://skypilot.readthedocs.io/en/latest/reference/yaml-spec.html ?

Reply:
Good point! Added. Thanks!